refactor(agw): Minor improvements to the fabfiles

MoritzThomasHuebner opened the pull request:
## Summary

- Fixed some typos and docstring formatting.
- Added missing docstrings for fabric connections.

- Renamed variables to snake_case.
- Renamed function arguments `id` to `network_id`. `id` shadows an in-built name.
- Renamed a variable `hash` to `commit_hash`. `hash` shadows an in-built name.

- Eliminated an unused connection variable.
- Eliminated an unused function argument in `_run_integ_tests`.
- Eliminated a `list()` default value.

## Test Plan
- [x] Run `fab integ-test` locally
- [x] ~Run AGW Integ Tests on CI https://github.com/magma/magma/actions/runs/4190570919/jobs/7264107426 (On magma origin repo because it does not work on my branch for some reason).~ Sorry, I should have done CWF, see below
- [x] https://github.com/magma/magma/actions/runs/4191459973

MoritzThomasHuebner committed:
refactor(agw): Various improvements

- Fixed some typos
- Eliminated unused function argument in `_run_integ_tests`
- Added missing docstrings for fabric connections

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
refactor(agw): Various improvements

- Fixed some typos and docstring formatting
- Renamed variables to snake_case
- Eliminated a `list()` default value.
- Renamed function arguments `id` to `network_id`. `id` shadows an in-built name.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
refactor(agw): Various improvements

- Renamed a variable `hash` to `commit_hash`. `hash` shadows an in-built name.
- Fixed some typos and docstring formatting
- Added a missing docstring entry for a fabric connection.
- Eliminated an unused connection variable

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

mpfirrmann commented on cwf/gateway/fabfile.py:
```diff
@@ -199,12 +199,12 @@ def integ_test(
         )
     else:
         _run_integ_tests(
-            c, test_host, trf_host, tests_to_run, test_re, count,
```
This should be tested. I'm not sure if the connection needs to be parsed even if it is not explicitly called in the function and gets flagged as unused IDE.

mpfirrmann reviewed (approved):
LGTM. Some CWF tests are failing, but this is not related to the changes in the fabfile.

voisey commented on cwf/gateway/fabfile.py:
```diff
@@ -64,12 +64,12 @@ def integ_test(
 
-    test_host: The ssh address string of the machine to run the tests on
-        on. Formatted as "host:port". If not specified, defaults to the
+    test_host: The ssh address string of the machine to run the tests on.
```
There's also a `vagrant_setup` import at the top of this file that you can get rid of if you like

MoritzThomasHuebner replied:
I removed the unused import

voisey commented on cwf/gateway/fabfile.py:
```diff
@@ -219,6 +219,7 @@ def transfer_artifacts(
     """
     Fetches service logs from Docker and optionally gets core dumps
     Args:
+        c: Fabric connection
```
Is there a reason you didn't add this elsewhere, e.g. to the docstring of `integ_test`?

MoritzThomasHuebner replied:
Thanks for raising this, I was just going through things the IntelliJ linter was highlighting for me at first. Turns out the `integ_test` function was missing the "Args:" line, which made this not register as a list of arguments. I added the "Args:" and all the missing arguments to the docstrings. I think it's a bit of a stretch to fully document all arguments of all functions in the fabfiles at the moment, though.

voisey commented on feg/gateway/fabfile.py:
```diff
@@ -262,7 +262,7 @@ def __init__(
     def __init__(
         self,
-        plmn_ids: List[str] = list(),
+        plmn_ids: List[str] = None,
```
If it's a list then shouldn't the default value be `[]`? Otherwise I think it should be `Optional[List[str]]`

MoritzThomasHuebner replied:
Mutable defaults should always be avoided in Python: https://docs.python-guide.org/writing/gotchas/
I'll add the `Optional` here as well.

MoritzThomasHuebner replied:
Actually on second thought, I will not add the `Optional` here since it is used nowhere else in the file. This might eventually be worth doing in a bigger refactoring PR.

voisey replied:
Sure, sounds good to me.

voisey commented on lte/gateway/fabfile.py:
```diff
@@ -247,7 +247,7 @@ def federated_integ_test(
 
     if orc8r_on_vagrant:
-        # Modify Vagrantfile to allow access to Orc8r running inside Vagrant
+        # Modify Vagrantfile to allow access to orc8r running inside Vagrant
```
I believe "Orc8r" is the convention for the way to write the component, see e.g. https://github.com/magma/magma/wiki/Contributing-Documentation#styling-and-writing-conventions in the documentation guidelines. Is there a reason to change it here?

MoritzThomasHuebner replied:
I changed this back

MoritzThomasHuebner committed:
refactor(agw): Improved fabfile docs

- Some Arguments were missing
- Added "Args:" pattern where missing
- Changed some formatting

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
refactor(agw): Changed the case of a letter

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

MoritzThomasHuebner committed:
refactor(agw): Removed an unused import

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

wolfseb reviewed (approved):
LGTM
